prov/verbs/src/ep_rdm/verbs_ep_rdm.c

```diff
@@ -163,11 +167,44 @@ static int fi_ibv_rdm_ep_bind(struct fid *fid, struct fid *bfid, uint64_t flags)
 		break;
 	case FI_CLASS_AV:
 		av = container_of(bfid, struct fi_ibv_av, av.fid);
+		if (ep->domain != av->domain) {
+			return -FI_EINVAL;
+		}
+
 		ep->av = av;
 
 		/* TODO: this is wrong, AV to EP is 1:n */
 		ep->av->ep = ep;
 		break;
+	case FI_CLASS_CNTR:
+		cntr = container_of(bfid, struct fi_ibv_rdm_cntr, fid.fid);
+		if (ep->domain != cntr->domain) {
+			return -FI_EINVAL;
+		}
+
+		if ((flags & FI_REMOTE_READ) || (flags & FI_REMOTE_WRITE)) {
+			return -FI_ENOSYS;
+		}
+
+		if (flags & FI_SEND) {
+			ep->send_cntr.cntr = cntr;
+			dlist_insert_tail(&ep->send_cntr.bind, &cntr->ep_list);
```

Comment:
BTW, utils provider does not have fi_cntr implementation, may be move this one as an initial draft for utils?